apps/omg_utils/lib/omg_utils/metrics.ex

```diff
@@ -33,18 +33,61 @@ defmodule OMG.Utils.Metrics do
   def measure_start(body, context) do
     # NOTE: the namespace and event group naming convention here is tentative.
     # It is possible we'll revert to standard coarser division into `web` and `background` namespaces Appsignal suggests
-    namespace = context.module |> Module.split() |> List.last() |> String.to_existing_atom()
-
-    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace)),
+    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace(context))),
       do: body,
-      else: Appsignal.Instrumentation.Decorators.transaction(namespace, body, context)
+      else: start_trace(body, context)
```

Comment:
hm, I read my old comment and noticed there are some decorators (similar to those of AppSignal) available in `Spandex`. Did they not work? I known that the macros you did might be more inline with the `telemetry` plan, but just curious why not use the decorators of `Spandex`?

Reply:
The decorators included with spandex produced a massive amount of `unmatched_return` dialyzer warnings (see build [here](https://circleci.com/gh/omisego/elixir-omg/11069?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)). 

Instead of ignore the warnings, it seemed like maybe a good idea to just write our own decorators since we're already doing that here to filter discarded namespace metrics. I copied the spandex decorator implementation (see it [here](https://github.com/spandex-project/spandex/blob/master/lib/decorators.ex)), but fixed the `unmatched_return` for our dialyzer warnings. As you mentioned, this also seemed useful since we want to put this in telemetry eventually.